### PR TITLE
Another upstream sync

### PR DESCRIPTION
**Pull request by travees**
Few more upstream changes before PR of OT specific changes

### CODE REVIEW

File: templates/conf/quorum.conf.erb [hunk 1]
```diff
@@ -1,7 +0,0 @@
-# quorum conf
```
Comment (SegFaultAX) [inline templates/conf/quorum.conf.erb:1]: Is this file unused now?

File: templates/conf/zoo.cfg.erb [hunk 2]
```diff
@@ -35,13 +35,13 @@ clientPortAddress=<%= @client_ip %>
...
-    <% observer_text=':observer' -%>
-  <% end -%>
-<%= "server.#{i+1}=#{h}:%s:%s%s" % [ @election_port, @leader_port, observer_text ] %>
+<% if h.index(':') -%>
+<% h = h[0...(h.index(':'))] -%>
+<% end -%>
+<% if @observers.include? h -%>
+<% observer_text=':observer' -%>
+<% end -%>
+<%= "server.#{i+1}=#{h}:%s:%s%s" % [ @election_port, @leader_port, observer_text ] -%>
```
Comment (SegFaultAX) [inline templates/conf/zoo.cfg.erb:44]: Is all this whitespace triggering you?

Comment (SegFaultAX) [inline templates/conf/zoo.cfg.erb:44]: Whoever wrote this Ruby code is also a weirdo. :D

**Comment by SegFaultAX** [comment]
:shipit: 💩 :trollface:

File: templates/conf/quorum.conf.erb (hunk 1 above)
Comment (travees) [inline templates/conf/quorum.conf.erb:1]: Looks like this config is incorporated into zoo.cfg
These commits are upstreams commits.

File: templates/conf/zoo.cfg.erb (hunk 2 above)
Comment (travees) [inline templates/conf/zoo.cfg.erb:44]: It does

**Pull request merged**